.github/workflows/build_and_test_pr.yml

```diff
@@ -0,0 +1,195 @@
+name: Pull Request Testing
+
+on:
+  pull_request:
+    branches: [ develop ]  # run this on any PR pointing to develop
+
+env:
+  GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  FC: gfortran-13
+  Python_REQUIRED_VERSION: 3.12.3  # 3.12.2 not available on Ubuntu 24 GHA
+  shell: bash
+
+jobs:
+  build_and_test:
+    runs-on: ${{ matrix.os }}
+    permissions:
+      pull-requests: write
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
```

Comment:
Full matrix of runs, note that one of each platform type has `run_regressions: true`.